common/scala/src/main/scala/whisk/core/entity/Exec.scala

```diff
@@ -47,37 +50,55 @@ import java.util.Base64
 sealed abstract class Exec(val kind: String) {
     def image: String
 
+    /**
+     * Method to check if the action does not exceed the sizeLimit, defined in Exec.sizeLimit.
+     * The size of the action is defined as the sum of sizes of everything, that is stores in the Exec(code, init, jar, main, ...).
+     *
+     * @return true if the exec has less characters than the sizelimit.
+     */
+    def isInSizeLimit: Boolean
+
     override def toString = Exec.serdes.write(this).compactPrint
 }
 
 protected[core] case class NodeJSExec(code: String, init: Option[String]) extends Exec(Exec.NODEJS) {
     val image = "whisk/nodejsaction"
+    def isInSizeLimit = (code sizeInBytes) + (init.getOrElse("") sizeInBytes) <= Exec.sizeLimit
 }
 
 protected[core] case class NodeJS6Exec(code: String, init: Option[String]) extends Exec(Exec.NODEJS6) {
     val image = "whisk/nodejs6action"
+    def isInSizeLimit = (code sizeInBytes) + (init.getOrElse("") sizeInBytes) <= Exec.sizeLimit
 }
 
 protected[core] case class PythonExec(code: String) extends Exec(Exec.PYTHON) {
     val image = "whisk/pythonaction"
+
+    def isInSizeLimit = (code sizeInBytes) <= Exec.sizeLimit
 }
 
 protected[core] case class SwiftExec(code: String) extends Exec(Exec.SWIFT) {
     val image = "whisk/swiftaction"
+    def isInSizeLimit = (code sizeInBytes) <= Exec.sizeLimit
 }
 
 protected[core] case class Swift3Exec(code: String) extends Exec(Exec.SWIFT3) {
     val image = "whisk/swift3action"
+    def isInSizeLimit = (code sizeInBytes) <= Exec.sizeLimit
 }
 
 protected[core] case class JavaExec(jar: String, main: String) extends Exec(Exec.JAVA) {
     val image = "whisk/javaaction"
+    def isInSizeLimit = (jar sizeInBytes) + (main sizeInBytes) <= Exec.sizeLimit
 }
 
-protected[core] case class BlackBoxExec(image: String) extends Exec(Exec.BLACKBOX)
+protected[core] case class BlackBoxExec(image: String) extends Exec(Exec.BLACKBOX) {
+    def isInSizeLimit = (image sizeInBytes) <= Exec.sizeLimit
+}
 
 protected[core] case class SequenceExec(code: String, components: Vector[String]) extends Exec(Exec.SEQUENCE) {
     val image = "whisk/nodejsaction"
+    def isInSizeLimit = (code sizeInBytes) + components.map(_ sizeInBytes).reduce(_ + _) <= Exec.sizeLimit
```

Comment:
Have a look at the implementation of `sum` on collections. It should be possible to provide that for `Size` aswell. Would make the code more readable.